src/Type/Php/ClassImplementsFunctionReturnTypeExtension.php

```diff
@@ -0,0 +1,62 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Type\Constant\ConstantBooleanType;
+use PHPStan\Type\DynamicFunctionReturnTypeExtension;
+use PHPStan\Type\ObjectWithoutClassType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+use PHPStan\Type\UnionType;
+use function count;
+use function in_array;
+
+class ClassImplementsFunctionReturnTypeExtension implements DynamicFunctionReturnTypeExtension
+{
+
+	public function isFunctionSupported(FunctionReflection $functionReflection): bool
+	{
+		return in_array(
+			$functionReflection->getName(),
+			['class_implements', 'class_uses', 'class_parents'],
+			true,
+		);
+	}
+
+	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): ?Type
+	{
+		if (count($functionCall->getArgs()) < 1) {
+			return null;
+		}
+
+		$objectOrClassType = $scope->getType($functionCall->getArgs()[0]->value);
+		$autoload = !isset($functionCall->getArgs()[1])
```

Comment:
`$autoload` should be a TrinaryLogic. So you should do `(new ConstantBooleanType(true))->isSuperTypeOf($secondArg)`. Because you might have `true` (yes), `bool` or `mixed` (maybe), or `false` (no).

Reply:
Why ?
If I'm not sure that `true` is passed, I cannot remove the `false` return type.
So `bool`, `mixed` or `false` (ie, maybe and no) are working the same way.

Reply:
Please test these scenarios.